File: 2050-count-good-numbers/count-good-numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        odd = n // 2
        even = (n + 1) // 2
        mod = pow(10, 9) + 7
        result = (pow(5, even, mod)  * pow(4, odd, mod)) % mod
        return result
        

File: 2050-count-good-numbers/count-good-numbers.java
class Solution {
    static final int MOD = 1000000007;
    public int countGoodNumbers(long n) {
        long even = (n+1)/2;
        long odd = n/2;

        long even_comb = powerMod(5, even);
        long odd_comb = powerMod(4, odd);

        return (int) ((even_comb * odd_comb) % MOD);
    }

    long powerMod(long base, long power) {
        long result = 1;
        base %= MOD;

        while(power > 0) {
            if (power % 2 == 0) {
                base = (base * base) % MOD;
                power /= 2;
            }
            else {
                result = (result * base) % MOD;
                power -= 1;
            }
        }
        return result;

    }
}